equinix/resource_network_device.go

```diff
@@ -570,6 +618,69 @@ func createNetworkDeviceSchema() map[string]*schema.Schema {
 				},
 			},
 		},
+		networkDeviceSchemaNames["ClusterDetails"]: {
+			Type:        schema.TypeList,
+			Optional:    true,
+			ForceNew:    true,
+			MaxItems:    1,
+			Description: networkDeviceDescriptions["ClusterDetails"],
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					neDeviceClusterSchemaNames["ClusterId"]: {
+						Type:        schema.TypeString,
+						Computed:    true,
+						Description: neDeviceClusterDescriptions["ClusterId"],
+					},
+					neDeviceClusterSchemaNames["ClusterName"]: {
+						Type:         schema.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringLenBetween(3, 50),
+						Description:  neDeviceClusterDescriptions["ClusterName"],
+					},
+					neDeviceClusterSchemaNames["NumOfNodes"]: {
+						Type:        schema.TypeInt,
+						Computed:    true,
+						Description: neDeviceClusterDescriptions["NumOfNodes"],
+					},
+					neDeviceClusterSchemaNames["ClusterNodeDetails"]: {
+						Type:        schema.TypeList,
+						Required:    true,
+						MaxItems:    1,
+						Description: neDeviceClusterDescriptions["ClusterNodeDetails"],
+						Elem: &schema.Resource{
+							Schema: map[string]*schema.Schema{
+								neDeviceClusterNodeSchemaNames["Node0"]: {
+									Type:     schema.TypeList,
+									Required: true,
+									MaxItems: 1,
+									Elem: &schema.Resource{
+										Schema: createClusterNodeDetailRequestSchema(),
```

Comment:
Let's merge the `ClusterNodeDetails ` and `Nodes` into one object.

The `createClusterNodeDetailRequestSchema` and `createNetworkDeviceClusterNodeSchema` schemas would be combined.

This will have the advantage of the vendor config block being defined and accessed through one attribute. 

I understand why we implemented this with two fields in this PR initially. We generally want to have Terraform match the API format. In this case, the request and response offer the same object with a different format, one as a named map entity, and the other as an array member. We can quell this difference for a simpler Terraform interface.

Reply:
The net result would be:
```hcl
resource "equinix_network_device" "foo" {
   cluster_details {
     cluster_node_details {
        node0 { ... }
        node1 { ... }
     }
   }
}
```

Users would access the hostname as:  `equinix_network_device.foo.cluster_details.cluster_node_details[0].node0[0].vendor_config.hostname`

Perhaps we should use the response key (`node`) instead of the request key (`cluster_node_details`).

Reply:
After some exploration and discussion with @rling-equinix on what the merged request and response HCL object should look like, we've arrived at a few different approaches that we could take here that would simplify the structure and access pattern:

```hcl
resource "equinix_network_device" "foo" {
    cluster_detail {
        cluster_name = "foo"

        # flatten the list of cluster_node_details, cluster_node_details only holds nodes, so hoist the nodes to the higher level
        # also considered naming them primary and secondary but this would be confused with the existing HA fields.
        node0 {} 
        node1 {}
        # user can access the COMPUTED node attributes with:
        # equinix_network_device.foo.cluster_detail.0.node0.0.name

        # Nodes with node elements
        nodes { # or name this cluster_node_details
            node0 {}
            node1 {}
        }
        # user can access the COMPUTED node attributes with:
        # equinix_network_device.foo.cluster_detail.0.nodes.0.node0.0.name

        # Nodes as array (singular name, order matters)
        node {
            vendor_config {}
        }

        node {
            vendor_config {}
        }
        # user can access the COMPUTED node attributes with:
        # equinix_network_device.foo.cluster_detail.0.nodes.0.name
    }
}
```

Reply:
Has there been any agreement here? Will it be merged? @rling-equinix

Reply:
After discussion with @rling-equinix @displague, we find in the option below the best solution to ensure that users always define both required nodes.

>         # flatten the list of cluster_node_details, cluster_node_details only holds nodes, so hoist the nodes to the higher level
>         # also considered naming them primary and secondary but this would be confused with the existing HA fields.
>         node0 {} 
>         node1 {}